Playbooks/playbook-DomainTools_Guided_Pivot_Enrichment.yml

```diff
@@ -0,0 +1,616 @@
+id: DomainTools_Guided_Pivot_Enrichment
```

Comment:
> What does Identity.TechnicalContact.Email.value} represent? What is the purpose of checking whether the .count is less than 500? Referring to task #4 in the playbook.

The value is the actual email address we want to pivot on. The count is how many other domains share that e-mail in our dataset. The check for less than 500 gives us a good pivot point threshold that gives us back domains that share this somewhat connected data attribute depending on what you pivot on.

Reply:
> The task "SetIndicatorTableData" should be removed, as explained earlier. Referring to task #2.

We are setting custom fields in the indicator table using this task. Is there a way to do that with the auto extractor?

Reply:
> Please divide the inputs so it can be used with any data (and not just from domaintools). So rather than having the input of ${Domain.DomainTools}, let's add individual, generic inputs, such as Email.Address, IP.Address and so on.

I'm not sure I can make this more generic as the domaintools data is needed because it contains the counts and actual values.

Reply:
I further explained my point about setting the indicator data manually and auto-extract, above. If you still have questions regarding that, feel free to ask here.

Making this playbook generic is very much possible, I'll explain how it works.:

Once you change the inputs to the generic ones as I showed, by doing this, amongst other changes (the rest of the inputs should be added too):
![image](https://user-images.githubusercontent.com/43602124/67283348-2d35e500-f4dc-11e9-9695-cb8998b6a911.png)

then you will have to modify your tasks to utilize the data from the input, as follows:
![image](https://user-images.githubusercontent.com/43602124/67283421-4e96d100-f4dc-11e9-9e3c-5db00158891b.png)
![image](https://user-images.githubusercontent.com/43602124/67283557-90277c00-f4dc-11e9-97ef-3a74fde8d7f6.png)

At this point you have a good flow of playbook inputs going into task inputs.
Now, how do we make the playbook receive the inputs we want it to?
Well, the inputs of any playbook **can be modified** to fit the place where they are used. Let's use your test playbook as an example. In the following pictures you will see on the left - the generic inputs I set for that playbook, and on the right - the modification I did to the inputs in the context of a subplaybook.
![image](https://user-images.githubusercontent.com/43602124/67284281-fe207300-f4dd-11e9-997c-ab2093046bac.png)

What I did was change the inputs of the subplaybooks only when used within the context of my main playbook.

Reply:
I guess I'm still confused by the use of generic inputs in this playbook. If I used generic inputs, the playbook wouldn't do anything because it wouldn't know when to pivot. I see they can be modified, but the playbook makes no sense without the use of the counts available through DomainTools context data.

Reply:
One of the inputs could be a hardcoded "IPAddressCount" / "IPPivotThreshhold" which will be 500. Then your task could use this playbook input.
Imagine someone wants to use this playbook for pivoting on indicators that he got from another integration - then you will have nothing under `${Domain.DomainTools}`, and nothing in your playbook input `domaintools_data` . To make it generic, the tasks should take the indicators from the playbook inputs, and the playbook inputs should take the indicators from standard context paths (Email.Address, IP.Address, File.MD5 and so on). 
Finally, when a playbook wants to use the guided pivot playbook, it will call it and modify its inputs to take the IP Addresses, URLs, Domains and so on, from wherever they want.
For example, Playbook A has `
domaintools_data.IPAddresses.address.value` in context. Playbook A calls Guided Pivot, so it will have to modify the inputs of Guided Pivot to take `
domaintools_data.IPAddresses.address.value` and put it under its `IPAddress` input. Guided Pivot will then pass `inputs.IPAddress` into its "IP Pivot" task.
Does the flow make more sense to you now? If there are still any hesitations or questions, ask me :)

Reply:
I think I understand now. Thanks for your patience in explaining this.